backend/src/jv2backend/journalRoutes.py

```diff
@@ -0,0 +1,187 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+# Copyright (c) 2023 Team JournalViewer and contributors
+
+"""Defines the Flask endpoints that only access journal information"""
+import logging
+from flask import Flask, jsonify, request
+from flask.wrappers import Response as FlaskResponse
+
+from jv2backend.journalClasses import JournalLibrary, JournalCollection
+from jv2backend.io.journals.networkLocator import NetworkJournalLocator
+from jv2backend.utils import json_response, split, url_join
+
+
+def add_routes(
+    app: Flask,
+    networkJournalLocator: NetworkJournalLocator,
+    journalLibrary: JournalLibrary
+) -> Flask:
+    """Add routes to the given Flask application."""
+
+    # ---------------- Queries ------------------
+    @app.post("/journals/list")
+    def listJournals() -> FlaskResponse:
+        """Return the list of journal files in a specified location
+
+        The POST data should contain:
+               rootUrl: The root network or disk location for the journals
+             directory: The directory in rootUrl to probe for journals
+            index_file: Name of the index file in the directory, if known
+
+        :return: A JSON response containing available journals in a
+                 list(BasicJournalFile), or an error
+        """
+        data = request.json
+        rootUrl = data["rootUrl"]
+        directory = data["directory"]
+        dataDirectory = data["dataDirectory"]
+
+        logging.debug(f"Listing journals for {directory} in {rootUrl}")
+
+        # If we already have a library collection for the specified
+        # rootUrl/directory, just return it
+        library_key = url_join(rootUrl, directory)
+        if library_key in journalLibrary:
+            logging.debug(
+                f"Returning existing journal collection for {library_key}")
+            return journalLibrary[library_key].to_basic()
+
+        try:
+            if (rootUrl.startswith("http")):
+                if "indexFile" in data:
+                    journalLibrary[library_key] = JournalCollection(
+                        networkJournalLocator.get_index(
+                            server_root=rootUrl, journal_directory=directory,
+                            index_file=data['indexFile'],
+                            data_directory=dataDirectory))
+                    return jsonify(journalLibrary[library_key].to_basic())
```

Comment:
```suggestion
                    return jsonify(journalLibrary[library_key].to_basic()), 200, {'ContentType':'application/json'}
```
Best to return the HTTP status code as well as the content type header